packages/panzer/disc-fe/src/evaluators/Panzer_ProjectToFaces_impl.hpp

```diff
@@ -183,126 +114,24 @@ void panzer::ProjectToFaces<EvalT, Traits>::
 evaluateFields(typename Traits::EvalData workset)
 { 
 
-  // The coefficients being calculated here in the projection to the face basis
-  // are defined as the integral over the face of the field dotted with the face
-  // normal vector. For a first-order face basis, single point integration is
-  // adequate, so the cubature here just provides the proper weighting.
-  // For higher order, a distinction between "cell" and Gauss points will need
-  // to be made so the field is appropriately projected.
-  const shards::CellTopology & parentCell = *basis->getCellTopology();
-  Intrepid2::DefaultCubatureFactory quadFactory;
+  // Restricting HDiv field, multiplied by the normal to the faces, into HVol on the faces.
+  // This code assumes affine mapping and the projection into 1 quadrature point for each face,
+  // which is identified with the face. This makes sense only for low order bases, for which
+  // HVol is constant
 
-  // Fast Method: One point quadrature on hex mesh. Assumes that the
-  // face area can be computed from two edge normals. This is only
-  // true if the mesh is square or rectangular. Will not work for
-  // paved meshes.
-  if (use_fast_method_on_rectangular_hex_mesh){
-    Teuchos::RCP< Intrepid2::Cubature<PHX::exec_space,double,double> > faceQuad;
-
-    // Collect the reference face information. For now, do nothing with the quadPts.
-    const unsigned num_faces = parentCell.getFaceCount();
-    std::vector<double> refFaceWt(num_faces, 0.0);
-    for (unsigned f=0; f<num_faces; f++) {
-      faceQuad = quadFactory.create<PHX::exec_space,double,double>(parentCell.getCellTopologyData(2,f), 1);
-      const int numQPoints = faceQuad->getNumPoints();
-      Kokkos::DynRankView<double,PHX::Device> quadWtsFast("quadWts",numQPoints);
-      Kokkos::DynRankView<double,PHX::Device> quadPtsFast("quadPts",numQPoints,num_dim);
-      faceQuad->getCubature(quadPtsFast,quadWtsFast);
-      for (int q=0; q<numQPoints; q++)
-        refFaceWt[f] += quadWtsFast(q);
-    }
+  //TODO: make this work w/ high order basis
+  const int intDegree = basis->order();
+  TEUCHOS_ASSERT(intDegree == 1);
```

Comment:
@mperego, it looks like the earlier code here supported higher-order bases.  Do we need to make the new version support higher-order as well?

Reply:
@CamelliaDPG no, it was not supporting high order basis. I think it was only an attempt in that direction, but we would do it differently now, using projection. As a matter of fact the code assumes to have that only one evaluation point per face, which make sense only for constants, i.e. for low order basis.

Reply:
OK, sounds good!